parsons/databases/mysql/create_table.py

```diff
@@ -1,87 +1,28 @@
-import ast
+from parsons.databases.database.database import DatabaseCreateStatement
+import parsons.databases.mysql.constants as consts
+
 import petl
 import logging
 
 logger = logging.getLogger(__name__)
 
-# The following values are the minimum and maximum values for MySQL int
-# types. https://dev.mysql.com/doc/refman/8.0/en/integer-types.html
-SMALLINT_VALUES = [-32768, 32767]
-MEDIUMINT_VALUES = [-8388608, 8388607]
-INT_VALUES = [-2147483648, 2147483647]
-
-# Additional padding to add on to the maximum column width to account
-# for the addition of future data sets.
-VARCHAR_PAD = .25
-
 
-class MySQLCreateTable():
+class MySQLCreateTable(DatabaseCreateStatement):
 
     def __init__(self):
+        super().__init__()
 
-        pass
+        self.VARCHAR_PAD = consts.VARCHAR_PAD
+        self.COL_NAME_MAX_LEN = consts.COL_NAME_MAX_LEN
+        self.RESERVED_WORDS = []
 
+    # This is for backwards compatability
     def data_type(self, val, current_type):
-        # Evaluate the MySQL data type of a given value. Then return
-        # the data type of the value, while also weighing what the
-        # current type is.
-
-        # Stop if the current type is already a varchar
-        if current_type == 'varchar':
-            return current_type
-
-        # Convert to string to reevaluate data type. If the value is already
-        # a string, is None, or is empty, with raise and error. These should
-        # be considered varchars.
-        try:
-            t = ast.literal_eval(str(val))
-        except ValueError:
-            return 'varchar'
-        except SyntaxError:
-            return 'varchar'
-
-        # If a float, then end
-        if type(t) == float:
-            return 'float'
-
-        # If an int and current type is not a float, then determine type of int.
-        if type(t) == int and current_type != 'float':
-            # Make sure that it is a valid integer
-            if not self.is_valid_integer(val):
-                return 'varchar'
-            # Use smallest possible int type above TINYINT
-            if (SMALLINT_VALUES[0] < t < SMALLINT_VALUES[1]) and current_type not in ['int', 'bigint', 'mediumint']:  # noqa: E501
-                return 'smallint'
-            elif (MEDIUMINT_VALUES[0] < t < MEDIUMINT_VALUES[1]) and current_type not in ['int', 'bigint']:  # noqa: E501
-                return 'mediumint'
-            elif (INT_VALUES[0] < t < INT_VALUES[1]) and current_type not in ['bigint']:
-                return 'int'
-            else:
-                return 'bigint'
-
-        # Finally return the current type
-        else:
-            return current_type
+        return self.detect_data_type(val, current_type)
 
+    # This is for backwards compatability
```

Comment:
What sort of backwards compatibility?

Reply:
This method, `is_valid_integer`, was replaced by `DatabaseCreateStatement.is_valid_sql_num`. However, that would break any scripts that use the previous method. Since I didn't want to make those kinds of changes in this PR, I just re-implemented  `is_valid_integer` to use `DatabaseCreateStatement.is_valid_sql_num` under the hood.